tests/test_csc.py

```diff
@@ -1,139 +1,60 @@
-
-                for bad_command in commands:
-                    if bad_command in ("enable", "standby"):
-                        continue  # valid command in DISABLED state
-                    with self.subTest(bad_command=bad_command):
-                        cmd_attr = getattr(harness.remote, f"cmd_{bad_command}")
-                        with self.assertRaises(salobj.AckError):
-                            id_ack = await cmd_attr.start(timeout=BASE_TIMEOUT)
-
-                # send enable; new state is ENABLED
-                cmd_attr = getattr(harness.remote, "cmd_enable")
-                harness.remote.evt_summaryState.flush()
-                id_ack = await cmd_attr.start(timeout=BASE_TIMEOUT)
-                state = await harness.remote.evt_summaryState.next(flush=False,
-                                                                   timeout=BASE_TIMEOUT)
-                self.assertEqual(id_ack.ack, salobj.SalRetCode.CMD_COMPLETE)
-                self.assertEqual(id_ack.error, 0)
-                self.assertEqual(harness.csc.summary_state, salobj.State.ENABLED)
-                self.assertEqual(state.summaryState, salobj.State.ENABLED)
-
-                for bad_command in commands:
-                    if bad_command == "disable":
-                        continue  # valid command in ENABLE state
-                    with self.subTest(bad_command=bad_command):
-                        cmd_attr = getattr(harness.remote, f"cmd_{bad_command}")
-                        with self.assertRaises(salobj.AckError):
-                            id_ack = await cmd_attr.start(timeout=BASE_TIMEOUT)
-
-                # send disable; new state is DISABLED
-                cmd_attr = getattr(harness.remote, "cmd_disable")
-                # this CMD may take some time to complete
-                id_ack = await cmd_attr.start(timeout=30.)
-                self.assertEqual(id_ack.ack, salobj.SalRetCode.CMD_COMPLETE)
-                self.assertEqual(id_ack.error, 0)
-                self.assertEqual(harness.csc.summary_state, salobj.State.DISABLED)
-
-        asyncio.get_event_loop().run_until_complete(doit())
-
-
-if __name__ == '__main__':
-
-    stream_handler = logging.StreamHandler(sys.stdout)
-    logger.addHandler(stream_handler)
-
-    unittest.main()
+class CscTestCase(salobj.BaseCscTestCase, asynctest.TestCase):
+    def basic_make_csc(self, initial_state, config_dir, simulation_mode, **kwargs):
+        return csc.CSC(
+            initial_state=initial_state,
+            config_dir=config_dir,
+            simulation_mode=simulation_mode,
+        )
+
+    async def test_standard_state_transitions(self):
+        logging.info("test_standard_state_transitions")
+        async with self.make_csc(
+            initial_state=salobj.State.STANDBY, config_dir=None, simulation_mode=1
+        ):
+            await self.check_standard_state_transitions(
+                enabled_commands=(
+                    "start",
```

Comment:
I'm not sure you need these commands as they are standard transitions.

Reply:
Good point. I added them because they were in the original test but they indeed are not needed so I removed them again.